main.go

```diff
@@ -123,11 +127,89 @@ func main() {
 	// Namespace where the operator is installed
 	currentNamespace := env.ReadOrPanic(util.CurrentNamespace)
 
+	// Get trace and span IDs from environment variables
+	traceIDHex := os.Getenv("OTEL_TRACE_ID")
+	spanIDHex := os.Getenv("OTEL_PARENT_ID")
+	endpoint := os.Getenv("OTEL_EXPORTER_OTLP_ENDPOINT")
+
 	// Get a config to talk to the apiserver
 	cfg := ctrl.GetConfigOrDie()
 
+	log.Infof("Setting up tracing with ID: %s, Parent ID: %s, Endpoint: %s", traceIDHex, spanIDHex, endpoint)
+	traceCtx, tp, err := telemetry.SetupTracing(signalCtx, traceIDHex, spanIDHex, endpoint)
+	if err != nil {
+		log.Errorf("Failed to setup tracing: %v", err)
+		// Continue without tracing instead of failing
+		traceCtx = signalCtx
+	} else if tp != nil {
+		defer func(tp *sdktrace.TracerProvider, ctx context.Context) {
+			log.Info("Shutting down tracer provider to ensure all spans are exported")
+			// Give time for queued spans to be exported before shutdown
+			shutdownCtx, cancel := context.WithTimeout(ctx, 5*time.Second)
+			defer cancel()
+
+			if err := tp.Shutdown(shutdownCtx); err != nil {
+				log.Errorf("Error shutting down tracer provider: %v", err)
+			} else {
+				log.Info("Tracer provider successfully shut down")
+			}
+		}(tp, signalCtx)
+		log.Info("OpenTelemetry tracing successfully configured")
+	} else {
+		log.Warn("No tracer provider created, tracing will be disabled")
+	}
+
+	log.Infof("Starting spans with trace context: %+v", traceCtx)
+
+	opts := []trace.SpanStartOption{
+		trace.WithAttributes(
+			attribute.String("component", "operator"),
+			attribute.String("namespace", currentNamespace),
+			attribute.String("service.name", "mongodb-kubernetes-operator"),
+			// let's ensure that the root span follows the given parent span
+			attribute.String("trace.parent_id", spanIDHex),
+		),
+		trace.WithSpanKind(trace.SpanKindServer),
+	}
+
+	// Use a very distinctive name for the root span to make it easier to find
+	ctxWithSpan, operatorSpan := telemetry.TRACER.Start(traceCtx, "MONGODB_OPERATOR_ROOT", opts...)
+	log.Infof("Started root operator span with ID: %s in trace %s", operatorSpan.SpanContext().SpanID().String(), operatorSpan.SpanContext().TraceID().String())
+
+	// We need to manually flush the span when the program ends
+	defer func() {
+		// Add more context to the root span before ending it
+		operatorSpan.SetAttributes(
+			attribute.String("operator.shutdown", "true"),
+			attribute.String("operator.span.status", "complete"),
+			attribute.Bool("root", true),
+		)
+
+		log.Info("Ending root operator span and flushing it to exporter")
```

Comment:
The messages on Info level will appear in normal, regular logs for every customer. And it looks like this is not guarded by having otel enabled or not. 

Please review every log message and consider if the customer seeing it in the logs will know what is this about and not be confusing.

Reply:
all reworked to debug, thanks for the catch!